### PR TITLE
RemoteTimelineClient::delete_all() to use s3::delete_objects

### PR DESCRIPTION
**Pull request by shanyp**
## Problem
[#4325](https://github.com/neondatabase/neon/issues/4325)
## Summary of changes
Use delete_objects() method

### CODE REVIEW

**Comment by shanyp** [comment]
I think the next phase for this is changing https://github.com/neondatabase/neon/blob/c508b1b0080201b0308e78e0af9aa9d2db14b49d/pageserver/src/tenant/remote_timeline_client.rs#L888

To support batching all of ```UploadOp::Delete``` tasks in a new ```UploadOp::DeleteObjects``` task

File: pageserver/src/tenant/remote_timeline_client.rs [hunk 1]
```diff
@@ -866,10 +866,8 @@ impl RemoteTimelineClient {
                 "Found {} files not bound to index_file.json, proceeding with their deletion",
                 remaining.len()
             );
-            for file in remaining {
-                warn!("Removing {}", file.object_name().unwrap_or_default());
-                self.storage_impl.delete(&file).await?;
-            }
+            warn!("About to remove {} files", remaining.len());
+            self.storage_impl.delete_objects(&remaining).await?;
```
Comment (problame) [inline pageserver/src/tenant/remote_timeline_client.rs:870]: When reading this patch, I was starting to think about partial failure handling.
E.g., what happens in the following cases

1. `delete_objects([A, B, C])` but [`B`] does not exist on the remote
2. `delete_objects([A, B, C])` but we don't have permission to delete [`B`]
3. ???

I guess even before this patch, we were bailing out on first error, and I assume there's a higher-level retry mechanism.

So, feel free to dismiss this comment.

**Review by problame — APPROVED** [review]
In addition to my comment, did you think through the 1000 objects limit ?

> The request contains a list of up to 1000 keys that you want to delete. 

https://docs.rs/aws-sdk-s3/0.27.0/aws_sdk_s3/operation/delete_objects/builders/struct.DeleteObjectsFluentBuilder.html

File: pageserver/src/tenant/remote_timeline_client.rs (hunk 1 above)
Comment (shanyp) [reply pageserver/src/tenant/remote_timeline_client.rs:870]: when objects don't exist on the remote , S3 will return success 
and yes we bailed on the first error

**Comment by shanyp** [comment]
> In addition to my comment, did you think through the 1000 objects limit ?
> 
> > The request contains a list of up to 1000 keys that you want to delete.
> 
> https://docs.rs/aws-sdk-s3/0.27.0/aws_sdk_s3/operation/delete_objects/builders/struct.DeleteObjectsFluentBuilder.html

https://github.com/neondatabase/neon/pull/4463

**Review by awestover — APPROVED** [review]
looks good!

**Review by LizardWizzard — APPROVED** [review]

**Pull request merged**